src/main/fc/fc_init.c

```diff
@@ -237,12 +237,13 @@ void init(void)
 
     initEEPROM();
 
-    ensureEEPROMContainsValidData();
-    readEEPROM();
+    ensureEEPROMStructureIsValid();
```

Comment:
`bool readSuccess = isEEPROMStructureValid() && readEEPROM();` is probably cleaner, avoiding hidden reset

Reply:
Agreed. I will have to rewrite this section once #5985 has been merged, to get around the board info being reset when EEPROM_VERSION changes. I will refactor this then.